Gems/PhysX/Code/Source/HeightfieldColliderComponent.cpp

```diff
@@ -62,22 +63,79 @@ namespace PhysX
     {
     }
 
-    void HeightfieldColliderComponent::Activate()
+
+    void HeightfieldColliderComponent::OnAssetReady(AZ::Data::Asset<AZ::Data::AssetData> asset)
     {
-        const AZ::EntityId entityId = GetEntityId();
+        if (asset == m_bakedHeightfieldAsset)
+        {
+            m_bakedHeightfieldAsset = asset;
 
-        AzPhysics::SceneHandle sceneHandle = AzPhysics::InvalidSceneHandle;
-        Physics::DefaultWorldBus::BroadcastResult(sceneHandle, &Physics::DefaultWorldRequests::GetDefaultSceneHandle);
+            Physics::HeightfieldShapeConfiguration& configuration = static_cast<Physics::HeightfieldShapeConfiguration&>(*m_shapeConfig);
+            configuration = Utils::CreateBaseHeightfieldShapeConfiguration(GetEntityId());
 
-        m_heightfieldCollider =
-            AZStd::make_unique<HeightfieldCollider>(GetEntityId(), GetEntity()->GetName(), sceneHandle, m_colliderConfig, m_shapeConfig);
+            // Update material selection from the mapping
+            Physics::ColliderConfiguration* colliderConfig = m_colliderConfig.get();
+            Utils::SetMaterialsFromHeightfieldProvider(GetEntityId(), colliderConfig->m_materialSlots);
 
-        ColliderComponentRequestBus::Handler::BusConnect(entityId);
-        Physics::CollisionFilteringRequestBus::Handler::BusConnect(entityId);
+            Pipeline::HeightFieldAsset* heightfieldAsset = m_bakedHeightfieldAsset.Get();
+
+            bool minMaxHeightsMatch = AZ::IsClose(configuration.GetMinHeightBounds(), heightfieldAsset->GetMinHeight()) &&
+                AZ::IsClose(configuration.GetMaxHeightBounds(), heightfieldAsset->GetMaxHeight());
+
+            if (!minMaxHeightsMatch)
+            {
+                AZ_Warning(
+                    "PhysX",
+                    false,
+                    "MinMax heights mismatch between baked heightfield and terrain. Entity [%s]. "
+                    "Terrain [%0.2f, %0.2f], Asset [%0.2f, %0.2f]",
+                    GetEntity()->GetName().c_str(),
+                    configuration.GetMinHeightBounds(),
+                    configuration.GetMaxHeightBounds(),
+                    heightfieldAsset->GetMinHeight(),
+                    heightfieldAsset->GetMaxHeight());
```

Comment:
This should say "between baked heightfield and heightfield provider". Even though the only heightfield provider right now is terrain, we might have others in the future.